OpenRA.Mods.Common/Pathfinder/PathSearch.cs

```diff
@@ -46,13 +50,19 @@ public static IPathSearch FromPoint(IWorld world, IMobileInfo mi, IActor self, C
 				heuristic = DefaultEstimator(target)
 			};
 
+			search.isGoal = loc =>
+			{
+				var locInfo = search.Graph[loc];
+				return locInfo.EstimatedTotal - locInfo.CostSoFar == 0;
+			};
```

Comment:
Maybe have ```PathSearch``` to pass ```this``` as a parameter to the ```isGoal``` delegate? Seems odd to need to capture the search object so you can query the graph but maybe I'm being overly specific to this use-case.